### PR TITLE
add PVSystem.losses_parameters, add support for custom pvwatts losses parameters in ModelChain

### LINKED ISSUE #484: pvwatts loss model implementation with user defined input values
Hi, 

I am trying to use pvwatts loss model with ModelChain with user defined loss values instead of the default ones. 

I understand that for default values I just have to pass "pvwatts" string as the losses_model but I am not sure how can I link the loss model with user-specified values in the ModelChain function. I have tried all the possible ways that I can think of but still not sure how this will be implemented. 

Will appreciate if anyone can guide me to it!  

Thanks in advance! 

Regards

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #484
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests must pass on the TravisCI and Appveyor testing services.
 - [x] Code quality and style is sufficient. Passes ``git diff upstream/master -u -- "*.py" | flake8 --diff`` and/or landscape.io linting service.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.

Added tests to ensure that custom losses are now supported in ModelChain.

### CODE REVIEW

File: docs/sphinx/source/whatsnew/v0.6.0.rst [hunk 1]
```diff
@@ -5,6 +5,11 @@ v0.6.0 (___, 2018)
 
 API Changes
 ~~~~~~~~~~~
+* pvsystem.calcparams_desoto now requires arguments for each module model
+  parameter. (:issue:`462`)
+* Add losses_parameters attribute to PVSystem objects and remove the kwargs
+  support from PVSystem.pvwatts_losses. Enables custom losses specification
+  in ModelChain calculations. (:issue:`484`)
 * pvsystem.calcparams_desoto now requires arguments for each module model parameter.
```
Comment (cwhanse) [inline docs/sphinx/source/whatsnew/v0.6.0.rst]: delete this line (redundant, and yours is better)

File: pvlib/pvsystem.py [hunk 2]
```diff
@@ -119,8 +119,7 @@ def __init__(self,
                  modules_per_string=1, strings_per_inverter=1,
                  inverter=None, inverter_parameters=None,
                  racking_model='open_rack_cell_glassback',
-                 name=None,
-                 **kwargs):
+                 losses_parameters=None, name=None, **kwargs):
```
Comment (cwhanse) [inline pvlib/pvsystem.py]: Need an entry in the signature describing `losses_parameters`

File: pvlib/pvsystem.py [hunk 3]
```diff
@@ -99,6 +99,9 @@ class PVSystem(object):
     racking_model : None or string, default 'open_rack_cell_glassback'
         Used for cell and module temperature calculations.
 
+    losses_parameters : None, dict or Series, default None
+        Losses parameters as defined by PVWatts or other.
```
[the remark this answers is not in the document]
Comment (wholmgren) [reply pvlib/pvsystem.py]: I'm not opposed to adding some additional detail to this doc string, but I would like to do it in a model-independent way. pvlib currently only includes the pvwatts losses model, but this dict should be useful to future models. Also note the current text is similar to the ``module_parameters`` and ``inverter_parameters`` doc strings.

**Comment by wholmgren** [comment]
Any other comments on this ``PVSystem.losses_parameters`` PR? I'll merge in the next few days if there are no objections and we can resolve the outstanding issue regarding the doc string.

**Comment by cwhanse** [comment]
I'm OK merging now and opening a new issue to improve the doc string and creating an example of the loss model use.

**Pull request merged**